app/services/learning_units/display_user_learning_unit.rb

```diff
@@ -0,0 +1,21 @@
+module LearningUnits
+  class DisplayUserLearningUnit
+    def initialize(curriculum, current_user)
+      @curriculum = curriculum
+      @user = current_user
+    end
+
+    def user_learning_units
+      @curriculum.learning_units.map do |learning_unit|
+        {
+          learning_unit:,
+          user: @user,
+          completed_by_user: CompletedLearningUnit.find_by(
```

Comment:
Could be better to make a special query for this? For each learning_unit there is a query searching the completed learning unit.